### PR TITLE
preview: update for pdf redirection to media-files

### PR DESCRIPTION
**Pull request by 0einstein0**
Closes Issue https://github.com/zenodo/zenodo-rdm/issues/931 

- Redirect pdf preview to mirador if set in the metadata of pdf

Screenshot 2024-07-11 at 14 41 48

### CODE REVIEW

File: site/zenodo_rdm/previewer/mirador.py [hunk 1]
```diff
@@ -13,6 +13,15 @@
 from flask import current_app, render_template
 
 
+def is_pdf_previewable(file):
+    """Check if the file is a PDF that should be previewed with Mirador."""
+    return (
+        file.has_extensions(".pdf")
+        and file.data["metadata"]["previewer"] == "mirador"
+        and file.record._record.media_files.entries is not None
```
Comment (yashlamba) [inline site/zenodo_rdm/previewer/mirador.py:21]:
```suggestion
        and file.record._record.media_files.entries
```

File: site/zenodo_rdm/previewer/mirador.py [hunk 2]
```diff
@@ -22,12 +31,16 @@ def can_preview(file):
     # supported_extensions list needs . prefixed -
     preview_extensions = current_app.config["MIRADOR_PREVIEW_EXTENSIONS"]
     supported_extensions = ["." + ext for ext in preview_extensions]
+
+    if is_pdf_previewable(file):
+        return True
```
Comment (yashlamba) [inline site/zenodo_rdm/previewer/mirador.py:36]: question: why?

Comment (0einstein0) [reply site/zenodo_rdm/previewer/mirador.py:36]: as discusssed in IRL
see below, 
```
# Find a suitable previewer
    fileobj = PreviewFile(file_metadata, pid_value, record, url)
    for plugin in current_previewer.iter_previewers(
        previewers=[file_previewer] if file_previewer else None
    ):
        if plugin.can_preview(fileobj):
            return plugin.preview(fileobj)
```

https://github.com/inveniosoftware/invenio-app-rdm/blob/3ea347e913befc0d813bc1df6f99590731ff19b8/invenio_app_rdm/records_ui/views/records.py#L302

File: site/zenodo_rdm/previewer/mirador.py [hunk 3]
```diff
@@ -13,6 +13,15 @@
 from flask import current_app, render_template
 
 
+def is_pdf_previewable(file):
+    """Check if the file is a PDF that should be previewed with Mirador."""
+    return (
+        file.has_extensions(".pdf")
+        and file.data["metadata"]["previewer"] == "mirador"
```
Comment (yashlamba) [inline site/zenodo_rdm/previewer/mirador.py:20]: Should we do a get here? Could the metadata be None?

**Review by yashlamba — COMMENTED** [review]
Just one minor comment, rest looks good.

**Review by yashlamba — APPROVED** [review]

**Review by slint — APPROVED** [review]

**Pull request merged**